spack.yaml

```diff
@@ -3,40 +3,65 @@
 # It describes a set of packages to be installed, along with
 # configuration settings.
 spack:
-  definitions:
-  - compilers:
-    - intel@2021.10.0
-  - mpis:
-    - openmpi@4.1.5
-  - serial_packages:
-    - fortranxml@4.1.2
-  - packages:
-    - esmf@8.5.0+debug
-    - parallelio@2.6.2 build_type=RelWithDebInfo fflags='-qno-opt-dynamic-align -convert
-      big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model source'
-      cflags='-qno-opt-dynamic-align -fp-model precise -std=gnu99'
-    - netcdf-c@4.9.2 build_system=cmake build_type=RelWithDebInfo
-    - netcdf-fortran@4.6.1
-    - fms@2023.02 precision=64 +large_file +deprecated_io ~gfs_phys ~openmp ~quad_precision
-      build_type=RelWithDebInfo
-    - access-om3 +install_libraries
   specs:
-  - matrix:
-    - [$mpis]
-    - [$%compilers]
-  - matrix:
-    - [$packages]
-    - [$^mpis]
-    - [$%compilers]
-  - matrix:
-    - [$serial_packages]
-    - [$%compilers]
+    - access-om3@git.2024.04.0
+  packages:
+    # Main Dependencies
+    access-om3-nuopc:
+      require:
+        - '@git.0.3.0'
+
+    # Other Dependencies
+    esmf:
+      require:
+        - '@8.5.0'
+        # - '-fp-model=precise'
+    parallelio:
+      require:
+        - '@2.6.2'
+    netcdf-c:
+      require:
+        - '@4.9.2'
+        - build_system=cmake build_type=RelWithDebInfo
```

Comment:
Probably needs `^hdf5@develop-1.14` per https://github.com/COSIMA/access-om3/issues/153

Reply:
I think this means we need ^hdf5@1.14.4.3 then ?

Is there spack syntax for greater than or equal to 1.14.4.3

Reply:
My understanding is that the issue has been patched in affected spack hdf versions so we shouldn't need to do anything here. But should be checked.

Reply:
Oh I see ...


```
    # Patch needed for HDF5 1.14.3 to fix signaling FPE checks from triggering
    # at dynamic type system initialization. The type system's builtin types
    # were refactored in 1.14.3 and switched from compile-time to run-time
    # initialization. This patch suppresses floating point exception checks
    # that would otherwise be triggered by this code. Later HDF5 versions
    # will include the patch code changes.
    # See https://github.com/HDFGroup/hdf5/pull/3837
    patch("hdf5_1_14_3_fpe.patch", when="@1.14.3")
```

https://github.com/spack/spack/blob/5c59bb87a4d3ed5a190f9ef37b98cd91c91f020f/var/spack/repos/builtin/packages/hdf5/package.py#L191

So yes, its needs to be >1.14.3

Reply:
The FPE issue was introduced in 1.14.3 so I don't think there's a need to specify any version restriction here unless there's another reason for >1.14.3?

Reply:
Oh I get it!

Thanks : )

No change needed